src/types.rs

```diff
@@ -406,20 +429,45 @@ impl FromRedisValue for StreamClaimReply {
 
 impl FromRedisValue for StreamPendingReply {
     fn from_redis_value(v: &Value) -> RedisResult<Self> {
-        let parts: (usize, String, String, Vec<Vec<String>>) = from_redis_value(v)?;
-        let mut reply = StreamPendingReply::default();
-        reply.count = parts.0.to_owned() as usize;
-        reply.start_id = parts.1.to_owned();
-        reply.end_id = parts.2.to_owned();
-        for consumer in &parts.3 {
-            let mut info = StreamInfoConsumer::default();
-            info.name = consumer[0].to_owned();
-            if let Ok(v) = consumer[1].to_owned().parse::<usize>() {
-                info.pending = v;
+        let parts: (usize, Option<String>, Option<String>, Vec<Vec<String>>) = from_redis_value(v)?;
+        let count = parts.0.to_owned() as usize;
+
+        if count == 0 {
+            Ok(StreamPendingReply::Empty)
+        } else {
+            let mut result = StreamPendingData::default();
+
+            let start_id = match parts.1.to_owned() {
+                Some(start) => Ok(start),
+                None => Err(Error::new(
```

Comment:
Would it be cleaner to return the start value above (instead of wrapping a Result) and then add `return Error::new(...` for the None match? 

```
            let start_id = match parts.0.to_owned() {
                Some(v) => v,
                None => return Error::new(
                    ErrorKind::Other,
                    "IllegalState: Non-zero pending expects start id",
                ))
            };
```

Reply:
`from_redis_value` expects a `RedisResult` so you can't just return an `Error` - it needs to be wrapped into the `RedisResult` ie.

```
           let start_id = match parts.0.to_owned() {
                Some(v) => v,
                None => return Err(Error::new(
                    ErrorKind::Other,
                    "IllegalState: Non-zero pending expects start id",
                )))
            };
```

The `?` in the original code handles this for you - I'm not sure whether it's more idiomatic to use the fail / success type with a `?` or to throw a `return` in there. My inner functional-programmer says that `return` inside of a match is a bit gross, so I'm leaning towards the original code.